pkg/kubectl/cmd/util/factory_client_access.go

```diff
@@ -416,10 +417,28 @@ func (f *ring0Factory) SuggestedPodTemplateResources() []schema.GroupResource {
 
 func (f *ring0Factory) Printer(mapping *meta.RESTMapping, options printers.PrintOptions) (printers.ResourcePrinter, error) {
 	p := printers.NewHumanReadablePrinter(f.JSONEncoder(), f.Decoder(true), options)
-	printersinternal.AddHandlers(p)
+	f.AddDefaultHandlers(p)
 	return p, nil
 }
 
+// TODO(juanvallejo): change printersinternal.AddHandlers to printersinternal.GetDefaultHandlers and return
+// a *printers.HandlerEntry list. Handle any errors returned from generating this list.
```

Comment:
@fabianofranz working on a separate PR for this. Will expose [pkg/printers.handlerEntry](https://github.com/kubernetes/kubernetes/blob/master/pkg/printers/humanreadable.go#L48), and update [pkg/printers/internalversion.AddHandlers](https://github.com/kubernetes/kubernetes/blob/master/pkg/printers/internalversion/printers.go#L67) to return a slice of `*pkg/printers#HandlerEntry`.

This will allow a new factory method that returns the list of client-side registered printer handlers, [per the doc on printer refactor](https://docs.google.com/document/d/15uieHE_QnfF9eEj-AAbAvo3xBqfyu5x1v5Cghzy4ZDI/edit?ts=5978e6a6#heading=h.1l1gx9w54z6p).

Your thoughts and feedback are welcome on this